api/datastore/migrate_data.go

```diff
@@ -46,7 +53,21 @@ func (store *Store) MigrateData() error {
 		AuthorizationService:    authorization.NewService(store),
 	}
 
-	return store.connectionMigrateData(migratorParams)
+	// restore on error
+	err = store.connectionMigrateData(migratorParams)
+	if err != nil {
+		logrus.Errorf("While DB migration %v. Restoring DB", err)
+		// Restore options
+		options := BackupOptions{
+			BackupPath: backupPath,
+		}
+		err := store.restoreWithOptions(&options)
+		if err != nil {
+			logrus.Fatalf("Failed restoring the backup. Error %s", err)
+		}
+	}
+
+	return err
```

Comment:
Suggest we have go tests that simulate failure and then we can check that it is working as intended.

Also check the output of logs as they should make sense to an end user that they can see what has gone wrong and have some idea of how possibly they could fix it.